themes/default/theme/src/ts/docs-main.ts

```diff
@@ -0,0 +1,83 @@
+var docsMainNavToggleWrapper = $(".docs-main-nav-toggle-wrapper");
+var docsNavToggleIcon = $(".docs-nav-toggle-icon");
+
+$(window).on("resize", function () {
+    setDocsMainNavPosition();
+    setTableOfContentsVisibility();
+    setMainNavHeight();
+}).trigger('resize');
+
+$(window).on("scroll", function () {
+    setDocsMainNavPosition();
+}).trigger('resize');
+
+$(window).on("load", function() {
+    setDocsMainNavPosition();
+    setTableOfContentsVisibility();
+    setMainNavHeight();
+});
+
+(function (document, $) {
+    var docsToggle = $(".docs-nav-toggle");
+
+    docsToggle.on("click", function () {
+        docsMainNavToggleWrapper.toggleClass("docs-nav-show");
+        docsMainNavToggleWrapper.toggleClass("docs-nav-hide");
+        docsNavToggleIcon.toggleClass("close-docs-main-nav");
+        docsNavToggleIcon.toggleClass("open-docs-main-nav");
+        setTableOfContentsVisibility()
+    });
+})(document, jQuery);
+
+function setDocsMainNavPosition() {
+    if ($(this).width() < 1280) {
```

Comment:
do these screen size related changes need to be done using javascript? We typically don't do this using javascript and this can make it difficult to reason about when some of the responsive related styling is handled in css and some in javascript.  Also, don't need to block on this, we can update this after. Though, if there is a reason this needs to be done this way ignore me.

Reply:
i think the answer is yes, it needs to be done this way. but half way through building this i realized i could do the open and close nav in css only. but i think that will require more work so id rather do this in another pr though. but id love to zoom and talk it through.